GetDKAN/dkan_feedback
# Feedback review improvements

## Description

As a DKAN site manager or editor, I'd like the ability to moderate feedback in effective and user-friendly ways (including previewing the feedback content, and being able to quickly bulk-delete spam, or bulk-label feedback), so that I can prevent being overwhelmed by spam comments.

This PR includes the following feedback improvements:

- Added custom_pub module as a dependency.
- Added 'archived' option as a publishing option for feedback nodes.
- Feedback was removed from dkan_workflow process.
- Created separate admin UI with three tabs: unpublished (in `admin/content/feedback/unpublished`), published (in `admin/content/feedback`) and archived (in `admin/content/feedback/archived`).
- Added bulk operations for archiving/unarchiving feedback.

## Steps to test:

- To get this working in your local add the following piece of code in `src/make/drupal.make`:

```
dkan_feedback:
    download:
      type: git 
      url: https://github.com/NuCivic/dkan_feedback.git
      branch: feedback-improvements
    type: module
```

Then run `dktl make` and after all the dependencies are downloaded, enable dkan_feedback.
- Create a new feedback node as an anonymous user, the node will not appear in the main `/feedback` page unless it is published.
- Go to Content -> Feedback (`admin/content/feedback`):
  - [ ] you should get a list of approved feedback nodes.
  - [ ] in the bulk operations list you should get options for archiving, deleting and unpublishing feedback (they should all work).
- Go to Content -> Unpublished feedback (`admin/content/feedback/unpublished`):
  - [ ] you should get a list of unpublished feedback nodes,
  - [ ] the tab title should include the amount of unpublished feedback nodes.
  - [ ] in the bulk operations list you should get options for archiving, deleting and publishing feedback (they should all work).
- Go to Content -> Archived feedback (`admin/content/feedback/archived`), 
  - [ ] you should get a list of archived feedback nodes,
  - [ ] the tab title should include the amount of archived feedback nodes,
  - [ ] in the bulk operations list you should get options for deleting and unarchiving feedback (they should all work).
